src/main.rs

```diff
@@ -85,14 +85,31 @@ fn main() {
     let prog = args[0].clone();
     let matches = match Options::new()
                                 .optflag("h", "help", "")
+                                .optopt("y", "yacckind",
+                                        "Yacc variant to be parsed (default: Original)",
+                                        "Original|Eco")
                                 .parse(&args[1..]) {
         Ok(m) => m,
-        Err(f) => { usage(prog, f.to_string().as_str()); return }
+        Err(f) => usage(prog, f.to_string().as_str())
     };
 
-    if matches.opt_present("h") || matches.free.len() != 3 {
+    if matches.opt_present("h") {
         usage(prog, "");
-        return;
+    }
+
+    let yacckind = match matches.opt_str("y") {
+        None => YaccKind::Original,
+        Some(s) => {
+            match &*s.to_lowercase() {
+                "original" => YaccKind::Original,
+                "eco" => YaccKind::Eco,
+                _ => usage(prog, &format!("Unknown yacckind '{}'.", s))
```

Comment:
If you call this a "Yacc variant" in the help text then it would make sense to use "variant" rather than "kind" in the error message.